kura/org.eclipse.kura.core.keystore/src/main/java/org/eclipse/kura/core/keystore/FilesystemKeystoreServiceImpl.java

```diff
@@ -233,21 +234,38 @@ private boolean keystoreExists(String keystorePath) {
         return keystorePath != null && new File(keystorePath).isFile();
     }
 
-    private void createKeystore(String keystorePath) throws IOException {
+    private void createKeystore(KeystoreServiceOptions options)
+            throws IOException, KuraException, KeyStoreException, NoSuchAlgorithmException, CertificateException {
```

Comment:
Do we need to throw so many exception to log them immediately on the caller?
Does it make sense to catch and distinguish the different exceptions/causes?

Reply:
I've captured all exception in one generic and thrown it.